includes/data-events/connectors/ga4/class-ga4.php

```diff
@@ -21,13 +22,25 @@
  */
 class GA4 {
 
+	/**
+	 * The events being watched.
+	 *
+	 * @var array
+	 */
+	public static $watched_events = [
+		'reader_logged_in',
+		'reader_registered',
+	];
+
 	/**
 	 * Initialize the class and registers the handlers
 	 *
 	 * @return void
 	 */
 	public static function init() {
-		Data_Events::register_handler( [ __CLASS__, 'handle_reader_logged_in' ], 'reader_logged_in' );
+		foreach ( self::$watched_events as $event_name ) {
+			Data_Events::register_handler( [ __CLASS__, 'handle_' . $event_name ], $event_name );
```

Comment:
This feels like a very fragile implementation. If the `watched_events` are modified it will call for an undefined method.